spec/requests/errors_spec.rb

```diff
@@ -0,0 +1,47 @@
+RSpec.describe ErrorsController, type: :request do
+  def rails_respond_without_detailed_exceptions
```

Comment:
I found this useful tidbit here: https://medium.com/@edgar/rspec-how-to-have-rails-request-specs-handling-errors-like-production-a8c21ed0308a

Reply:
Whoa, this is really interesting. I gave that article a read and started clicking through to [the linked example](https://github.com/eliotsykes/rspec-rails-examples/issues/76) from @eliotsykes which led down a rabbit hole of associated issues, repos, and code examples.

I poked around a bit locally and came up with a possible small cleanup to this spec:

```rb
# spec/support/error_responses.rb

module ErrorResponses
  def rails_respond_without_detailed_exceptions
    env_config = Rails.application.env_config
    original_show_exceptions = env_config['action_dispatch.show_exceptions']
    original_show_detailed_exceptions = env_config['action_dispatch.show_detailed_exceptions']
    env_config['action_dispatch.show_exceptions'] = true
    env_config['action_dispatch.show_detailed_exceptions'] = false
    yield
  ensure
    env_config['action_dispatch.show_exceptions'] = original_show_exceptions
    env_config['action_dispatch.show_detailed_exceptions'] = original_show_detailed_exceptions
  end
end

RSpec.configure do |config|
  config.include ErrorResponses
end
```

…and…

```rb
# spec/requests/errors_spec.rb

RSpec.describe ErrorsController, type: :request do
  around :each do |example|
    rails_respond_without_detailed_exceptions(&example)
  end

  describe 'GET #not_found' do
    before do
      get '/page_that_doesnt_exist.pdf'
    end

    # same ol' specs here as before…
  end

  describe 'GET #internal_server_error' do
    before do
      expect(Entitlement).to receive(:all).and_raise('kaboom!')

      get '/entitlements'
    end

    # same ol' specs here as before…
  end
end
```

I moved the `rails_respond_without_detailed_exceptions ` method out into its own "support" file [based on this example](https://github.com/eliotsykes/rails-testing-toolbox/blob/master/error_responses.rb), wrapped it in a module, and included that module in Rspec's config. This file placement aligns with FactoryBot's and seems to be a pattern for globally including code in an Rspec test run (in case we want to test this error-handling behavior in other specs).

The `around` bit was grabbed [from this example](https://github.com/eliotsykes/rspec-rails-examples/issues/76#issuecomment-126937222) and cleans up some of the repetition in the `errors_spec.rb` file (since all specs in that file should use the error-handling patch from the support file).

Reply:
Just in case this article didn't turn up in your research, using the `:realistic_error_responses` metadata is an alternative to using the `around` technique above: https://www.wetestrails.com/blog/realistic-error-responses - HTH, Eliot

Reply:
@eliotsykes Ah, thanks for the link! I saw the linked code example [in your comment here](https://github.com/eliotsykes/rspec-rails-examples/issues/76#issuecomment-313720044), but didn't fully understand it. Looks like you've updated that and linked to the wetestrails.com post, as well.

Thanks for the update and for replying here!